sdk/src/client.ts

```diff
@@ -1,79 +1,32 @@
-import { WebSocket } from 'ws';
-import type { AuthStorage } from './composables/authentication.js';
-import { withoutTrailingSlash } from './utils.js';
-// import type { ReadItemsInput, ReadItemsOutput } from './commands/read-items.js';
-// import type { Command } from './types/index.js';
-
 export interface ClientConfig {
-	fetch?: fetch;
-	ws?: typeof WebSocket;
-	url?: string;
 	token?: string;
 }
 
-export interface BaseClientConfig extends Record<string, any> {
-	fetch: fetch;
-	ws: typeof WebSocket;
-	apiURL: string;
-}
-
-export interface GenericClient<Schema extends object, Features extends object> {
-	config: BaseClientConfig;
-	auth: AuthStorage;
-
-	use: <ExtraFeatures extends object>(
-		feature: (client: DirectusClient<Schema, Features>) => ExtraFeatures
-	) => DirectusClient<Schema, Features & ExtraFeatures>;
+export interface DirectusClient<TSchema extends object> {
+	config: { url: URL };
+	auth: { token: string | null };
+	use: <TExtension extends object>(
+		createExtension: (client: DirectusClient<TSchema>) => TExtension
+	) => this & TExtension;
 }
 
-// idk just a better type hint name xD
-export type BaseClient = Record<string, never>;
-
-// sometimes you know better than prettier
-// eslint-disable-next-line prettier/prettier
-export type DirectusClient<
-	Schema extends object = any,
-	Features extends object = BaseClient
-> = GenericClient<Schema, Features> & Features;
-
-// type InputTypes = ReadItemsInput<any>;
-// type OutputTypes = ReadItemsOutput<any, any>;
-
-export const useDirectus = <Schema extends object>(config: ClientConfig = {}) => {
-	// return a bare bones client
-	return {
+export const useDirectus = <TSchema extends object = any>(url: string, config?: ClientConfig) => {
+	const client: DirectusClient<TSchema> = {
 		config: {
-			fetch: config.fetch ?? fetch,
-			ws: config.ws ?? WebSocket,
-			apiURL: config.url ? withoutTrailingSlash(config.url) : '',
+			url: new URL(url),
```

Comment:
By relying on an URL object instead of a string, we should be able to reuse the baseURL between WS and REST/GQL. We can pull `host` from the URL, which omits the protocol. It also means that we don't have to manually do any string manipulation for URLs, which is bound to break somewhere somehow